src/ssh_audit/ssh_audit.py

```diff
@@ -836,82 +839,129 @@ def process_commandline(out: OutputBuffer, args: List[str], usage_cb: Callable[.
+    parser.add_argument('-l', '--level', action="store", dest='level', type=str, default='info')
+    parser.add_argument('-L', '--list-policies', action="store_true", dest='list_policies', default=None)
+    parser.add_argument('-M', '--make-policy', action="store", dest='make_policy', default=None)
+    parser.add_argument('-m', '--manual', action="store_true", dest='manual', default=None)
+    parser.add_argument('-P', '--policy', action="store", dest='policy', default=None)
+    parser.add_argument('-p', '--port', action="store", dest='port', default='22', type=int)
+    parser.add_argument('-T', '--targets', action="store", dest='targets', default=None)
+    parser.add_argument('-t', '--timeout', action="store", dest='timeout', default='5', type=int)
+    parser.add_argument('-v', '--verbose', action="store_true", dest='verbose', default=None)
+
+
+    # Add long options to the parser
+    parser.add_argument('--conn-rate-test', action="store", dest='conn_rate_test', default='0', type=int)
+    parser.add_argument('--dheat', action="store", dest='dheat', default='0', type=int)
+    parser.add_argument('--lookup', action="store", dest='lookup', default=None)
+    parser.add_argument('--no-colors', action="store_true", dest='no_colors', default=None)
+    parser.add_argument('--skip-rate-test', action="store_true", dest='skip_rate_test', default=None)
+    parser.add_argument('--threads', action="store", dest='threads', default='1', type=int)
+
+    try:
+        namespace = parser.parse_args()
+
+        if namespace.help is True:
             usage_cb(out)
-        elif o in ('-1', '--ssh1'):
-            aconf.ssh1 = True
-        elif o in ('-2', '--ssh2'):
-            aconf.ssh2 = True
-        elif o in ('-4', '--ipv4'):
-            aconf.ipv4 = True
-        elif o in ('-6', '--ipv6'):
-            aconf.ipv6 = True
-        elif o in ('-p', '--port'):
-            oport = a
-        elif o in ('-b', '--batch'):
+
+        aconf.host = namespace.host
+        host = namespace.host
+        port = namespace.port
+        aconf.ssh1 = namespace.ssh1
+        aconf.ssh2 = namespace.ssh2
+        aconf.ipv4 = namespace.ipv4
+        aconf.ipv6 = namespace.ipv6
+
+        aconf.json = namespace.json
+        if namespace.json_indent is True:
+            setattr(namespace, 'json', True)
+            aconf.json = namespace.json
+            aconf.json_print_indent = namespace.json_indent
+
+        if namespace.batch is True:
             aconf.batch = True
             aconf.verbose = True
-        elif o in ('-c', '--client-audit'):
-            aconf.client_audit = True
-        elif o in ('-j', '--json'):
-            if aconf.json:  # If specified twice, enable indent printing.
-                aconf.json_print_indent = True
-            else:
-                aconf.json = True
-        elif o in ('-v', '--verbose'):
+
+        aconf.client_audit = namespace.client_audit
+
+        ttime = namespace.timeout
+        if ttime != 5:
```

Comment:
Hmm... the original code didn't include this logic.  What was the rationale for this?

Reply:
**Note:** As has been seen I am not well versed with git / merging / comments .
As such the following comments have all been done manually ; so at least I know what I am doing.

The original getopts code has the following 

**Line ~879**
```
        elif o in ('-t', '--timeout'):
            aconf.timeout = float(a)
            aconf.timeout_set = True
```

**Line ~1200**

`s = SSH_Socket(out, aconf.host, aconf.port, aconf.ip_version_preference, aconf.timeout, aconf.timeout_set)`

This passes through and --timeout argument for the test to run.

During my cut over and testing for argparse I encountered several issues were the timeout was populated by a default value.

From memory the error code mentioned a 5 second default test timeout.

My argparse changes forces the the argument to a default value of 5 to match the behavior I observed; checks to see if the value has been modified and if so; sets the timeout value to that number as a float and sets timeout_set = True the same as the original script.